### PR TITLE
dns: T738: add CLI option for PowerDNS local-port

### PR DESCRIPTION
**Pull request by Zen3515**
## Change Summary
Add config which allow PowerDNS to run on different ports.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T738

## Component(s) name

pdns

## Proposed changes

Add cli option named `port` inside `service dns forwardning` which generate `local-port` inside pdns configuration

## How to test

```shell
set service dns forwarding port 5353
```

Expected conf:

```shell
vyos@r14# cat /run/powerdns/recursor.conf | grep -port
# listen-port
local-port=5353
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/dns-forwarding.xml.in [hunk 1]
```diff
@@ -605,6 +605,20 @@
                 </properties>
               </leafNode>
               #include <include/listen-address.xml.i>
+              <leafNode name="listen-port">
```
Comment (c-po) [inline interface-definitions/dns-forwarding.xml.in]: Please use `#include <include/port-number.xml.i>` instead. We try to have a consistent CLI for the same options and also re-use already existing CLI definitions (`include/port-number.xml.i`)

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/dns-forwarding.xml.in (hunk 1 above)
Comment (Zen3515) [reply interface-definitions/dns-forwarding.xml.in]: I use that at first, but I don't know how to override the default value when using the include. It needs to be 53, could you tell me how to set the default value?

Comment (sarthurdev) [reply interface-definitions/dns-forwarding.xml.in]: Like below, taken from snmp template

```
#include <include/port-number.xml.i>
<leafNode name="port">
  <defaultValue>161</defaultValue>
</leafNode>
```

Comment (Zen3515) [reply interface-definitions/dns-forwarding.xml.in]: Thank you, I've updated the commit, PTAL.

**Review by c-po — APPROVED** [review]

**Pull request merged**